src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -568,5 +580,17 @@ private void CheckSize(int countToAdd)
                 }
             }
         }
+
+        private struct EncodingPreamblePair
```

Comment:
Well, I checked the x86 results and it's funny, both struct and class generate good code yet the struct case is slower (1000ms vs 800ms). That's until you invert the order of the tests and then the struct case is slightly faster (730ms vs 770ms). Oh well, I guess we'll never get a realistic result by using such benchmarks. Anyone offers to flip a coin? :smile: